addons/calendar/models/calendar_event.py

```diff
@@ -930,7 +934,7 @@ def _split_recurrence(self, time_values):
             # When we try to change recurrence values of an event not following the recurrence, we get the parameters from
             # the base_event
             previous_week_day_field = weekday_to_field(self.recurrence_id.base_event_id._get_start_date().weekday())
-        self.write(time_values)
+        self.with_context(self._get_split_context()).write(time_values)
```

Comment:
with_context({key:val}) will override the existing context
with_context(key=val) will update the existing context with the ney key value.
we should probably hae something like that to avoid erasing the existing context: 
```py
return {**self._context, **{'dont_notify': True}}
```
or something similar. What do you think ?

Reply:
You are right! I tested it here and it was overriding the existing context, then we were losing keys such as 'lang', 'tz' 'uid' and 'allowed_company_ids'. 

Code updated, thanks for the tip! ‎😃